connection_manager.go

```diff
@@ -153,6 +154,26 @@ func (n *connectionManager) Run() {
 	}
 }
 
+// Check if peer's certificate is not expired or invalid.
+func (n *connectionManager) checkToDisconnect(now time.Time, hostinfo *HostInfo) (bool, *cert.NebulaCertificate) {
+	if !n.intf.disconnectInvalid {
+		return false, nil
+	}
+
+	if hostinfo == nil {
+		return false, nil
+	}
+
+	if remoteCert := hostinfo.GetCert(); remoteCert != nil {
+		valid, _ := remoteCert.Verify(now, n.intf.caPool)
```

Comment:
Passing the error back to the caller would allow the `disconnect` log to differentiate why (cert expired, root expired, could not find ca, etc)